cmd/device-worker/main.go

```diff
@@ -147,9 +151,45 @@ func main() {
 
 	setupSignalHandler(metricsStore)
 
+	go listenStartGracefulRebootChannel(err, wl, dataMonitor, systemMetricsWatcher, metricsStore, hbs, deviceOs)
+
 	if err := s.Serve(l); err != nil {
 		log.Fatalf("cannot start worker server, err: %v", err)
 	}
+
+}
+
+func listenStartGracefulRebootChannel(err error, wl *workload2.WorkloadManager, dataMonitor *datatransfer.Monitor, systemMetricsWatcher *metrics.SystemMetrics, metricsStore *metrics.TSDB, hbs *heartbeat2.Heartbeat, deviceOs *os2.OS) {
+	// listen to the channel for getting StartGracefulReboot signal
+	for {
+		select {
+		case job := <-GracefulRebootChannel:
+			if job == os2.StartGracefulReboot {
```

Comment:
is there an option for other values for this? else, we can just pick the type as of the channel as `struct{}{}`.